metpy/calc/tools.py

```diff
@@ -330,3 +332,190 @@ def log_interp(x, xp, fp, **kwargs):
         interpolated_vals = interpolated_vals * fp.units
 
     return interpolated_vals
+
+
+def _get_bound_pressure_height(pressure, bound, heights=None, interpolate=True):
+    """Calculate the bounding pressure and height in a layer.
+
+    Given pressure, optional heights, and a bound, return either the closest pressure/height
+    or interpolated pressure/height. If no heights are provided, a standard atmosphere is
+    assumed.
+
+    Parameters
+    ----------
+    pressure : `pint.Quantity`
+        Atmospheric pressures
+    bound : `pint.Quantity`
+        Bound to retrieve (in pressure or height)
+    heights : `pint.Quantity`
+        Atmospheric heights associated with the pressure levels
+    interpolate : boolean
+        Interpolate the bound or return the nearest
+
+    Returns
+    -------
+    `pint.Quantity`
+        The bound pressure and height.
+
+    """
+    # Bound is given in pressure
+    if bound.dimensionality == {'[length]': -1.0, '[mass]': 1.0, '[time]': -2.0}:
+        # If the bound is in the pressure data, we know the pressure bound exactly
+        if bound in pressure:
+            bound_pressure = bound
+            # If we have heights, we know the exact height value, otherwise return standard
+            # atmosphere height for the pressure
+            if heights is not None:
+                bound_height = heights[pressure == bound_pressure]
+            else:
+                bound_height = pressure_to_height_std(bound_pressure)
+        # If bound is not in the data, return the nearest or interpolated values
+        else:
+            if interpolate:
+                bound_pressure = bound  # Use the user specified bound
+                if heights is not None:  # Interpolate heights from the height data
+                    bound_height = log_interp(bound_pressure, pressure, heights)
+                else:  # If not heights given, use the standard atmosphere
+                    bound_height = pressure_to_height_std(bound_pressure)
+            else:  # No interpolation, find the closest values
+                idx = (np.abs(pressure - bound)).argmin()
+                bound_pressure = pressure[idx]
+                if heights is not None:
+                    bound_height = heights[idx]
+                else:
+                    bound_height = pressure_to_height_std(bound_pressure)
+
+    # Bound is given in height
+    elif bound.dimensionality == {'[length]': 1.0}:
+        # If there is height data, see if we have the bound or need to interpolate/find nearest
+        if heights is not None:
+            if bound in heights:  # Bound is in the height data
+                bound_height = bound
```

Comment:
What pressure error? Isn't that block just:
1. Bound in height
2. Passed in heights
3. Bound is one of the values in the heights array